scipy/linalg/tests/test_basic.py

```diff
@@ -931,40 +930,121 @@ def test_check_finite(self):
 
 class TestDet:
     def setup_method(self):
-        np.random.seed(1234)
-
-    def test_simple(self):
-        a = [[1, 2], [3, 4]]
-        a_det = det(a)
-        assert_almost_equal(a_det, -2.0)
-
-    def test_simple_complex(self):
-        a = [[1, 2], [3, 4j]]
-        a_det = det(a)
-        assert_almost_equal(a_det, -6+4j)
-
-    def test_random(self):
-        basic_det = linalg.det
-        n = 20
-        for i in range(4):
-            a = random([n, n])
-            d1 = det(a)
-            d2 = basic_det(a)
-            assert_almost_equal(d1, d2)
-
-    def test_random_complex(self):
-        basic_det = linalg.det
-        n = 20
-        for i in range(4):
-            a = random([n, n]) + 2j*random([n, n])
-            d1 = det(a)
-            d2 = basic_det(a)
-            assert_allclose(d1, d2, rtol=1e-13)
-
-    def test_check_finite(self):
-        a = [[1, 2], [3, 4]]
-        a_det = det(a, check_finite=False)
-        assert_almost_equal(a_det, -2.0)
+        self.rng = np.random.default_rng(1680305949878959)
+
+    def test_1by1_stacked_input_output(self):
+        a = self.rng.random([4, 5, 1, 1], dtype=np.float32)
+        deta = det(a)
+        assert deta.dtype.char == 'd'
+        assert deta.shape == (4, 5)
+        assert_almost_equal(deta, np.squeeze(a))
+
+        a = self.rng.random([4, 5, 1, 1], dtype=np.float32)*np.complex64(1.j)
+        deta = det(a)
+        assert deta.dtype.char == 'D'
+        assert deta.shape == (4, 5)
+        assert_almost_equal(deta, np.squeeze(a))
+
+    @pytest.mark.parametrize('shape', [[2, 2], [20, 20], [3, 2, 20, 20]])
+    def test_simple_det_shapes_real_complex(self, shape):
+        a = self.rng.uniform(-1., 1., size=shape)
+        d1, d2 = det(a), np.linalg.det(a)
+        assert_almost_equal(d1, d2)
+
+        b = self.rng.uniform(-1., 1., size=shape)*1j
+        b += self.rng.uniform(-0.5, 0.5, size=shape)
+        d3, d4 = det(b), np.linalg.det(b)
+        assert_almost_equal(d3, d4)
+
+    def test_for_known_det_values(self):
+        # Hadamard8
+        a = np.array([[1, 1, 1, 1, 1, 1, 1, 1],
+                      [1, -1, 1, -1, 1, -1, 1, -1],
+                      [1, 1, -1, -1, 1, 1, -1, -1],
+                      [1, -1, -1, 1, 1, -1, -1, 1],
+                      [1, 1, 1, 1, -1, -1, -1, -1],
+                      [1, -1, 1, -1, -1, 1, -1, 1],
+                      [1, 1, -1, -1, -1, -1, 1, 1],
+                      [1, -1, -1, 1, -1, 1, 1, -1]])
+        assert_almost_equal(det(a), 4096.)
+
+        # consecutive number array always singular
+        assert_almost_equal(det(np.arange(25).reshape(5, 5)), 0.)
+
+        # simple anti-diagonal array with known det
```

Comment:
Hmm this one and the next are not obvious to me, and a little googling didn't help. We are very liberal with adding references in stats to aid future review, but I will respect the customs of `linalg`.

Reply:
This one is just something I manually crafted, the next one is `scipy.linalg.fiedler_companion([1,...,9])` which has the determinant 9. They are nothing special in terms of fame.

Reply:
added descriptions regardless